test/block.spec.js

```diff
@@ -18,9 +18,9 @@ test('Block encode', done => {
 
 test('Block data caching', done => {
   const block = Block.encoder({ hello: 'world' }, 'dag-cbor')
-  const encoded = block.encode()
+  const encoded = block.encodeUnsafe()
```

Comment:
Are you testing here that when you're manipulating an object that is returned by `decodeUnsafe()`, that the original object is also changed, hence that manipulation shows up in the next `decode()` step?

If yes, then I don't think we should do this. It think such a manipulation should be undefined behaviour as mentioned in the README. I don't want to see people using this fact as a feature, it is just an implementation detail.

Reply:
I gather this test is just making assertions about caching, not that you can mutate, in which case it seems reasonable to me. An alternative might just be an identity check, `===`, which should say the same thing and doesn't demonstrate the kind of use the readme warns against.

Reply:
So, you **are definitely not suppose to do this when you use the library** ;) This test actually demonstrates the hazards of using the Unsafe methods, but it’s necessary in order to test that it is actually exposing the underlying cache object and not a copy like the safe methods do.